packages/devtools_app/lib/src/performance/legacy/performance_screen.dart

```diff
@@ -321,10 +321,10 @@ class LegacyPerformanceScreenBodyState
 
   @override
   bool shouldLoadOfflineData() {
-    return offlineMode &&
-        offlineDataJson.isNotEmpty &&
-        offlineDataJson[LegacyPerformanceScreen.id] != null &&
-        offlineDataJson[LegacyPerformanceData.traceEventsKey] != null;
+    return offlineController.offlineMode &&
```

Comment:
nit: seems like room for a helper checking that you are in offline mode and that the screen id namtches.

Reply:
done